src/components/checkbox/Checkbox.tsx

```diff
@@ -75,8 +78,9 @@ export const Checkbox = forwardRef<HTMLButtonElement, PropsWithLabels<Props>>((p
         <Spinner isLoading={isLoading}>
           <CheckboxRoot
             aria-disabled={isDisabled}
+            aria-invalid={hasError}
             aria-label={ariaLabel ?? title}
-            aria-labelledby={ariaLabelledBy}
+            aria-labelledby={ariaLabelledBy ?? labelId}
```

Comment:
Isn't adding the `labelId` here duplicating the `htmlFor` we have on the `Label` component?

Reply:
Yeah, but without it axe is complaining that there is no discernible text for the button, even though the button has the `checkbox` role, and a button is a labelable element.

```
"Buttons must have discernible text (button-name)"

Fix any of the following:
  Element does not have inner text that is visible to screen readers
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element has no title attribute
  Element's default semantics were not overridden with role="none" or role="presentation"
```

Do you think we should just get rid of the `for` attribute then?

Reply:
Ok so let's just get rid of the "for" and keep only the labelledBY

Reply:
If I remove the `for` attribute, then clicking the label does not check the checkbox ☹️

Reply:
Honestly, I think axe is probably wrong in this case

Reply:
I think this was fixed in axe-core [`v4.10.1`](https://github.com/dequelabs/axe-core/issues/3696). Unfortunately, jest-axe `v9.0.0` has a dependency on axe-core `v4.9.1`, and v9 is the latest version.

We can try using a resolution to use a new version of axe-core.

Reply:
If I add a resolution for `"axe-core": "4.10.2"` it fixes it.

Reply:
I can submit an upstream PR to bump axe-core in jest-axe and maybe they will merge it quickly